proxies/http_connect.go

```diff
@@ -0,0 +1,110 @@
+/*
+	Copyright NetFoundry Inc.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+	https://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+package proxies
+
+import (
+	"bufio"
+	"context"
+	"github.com/michaelquigley/pfxlog"
+	"github.com/pkg/errors"
+	"golang.org/x/net/proxy"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+func NewHttpConnectProxyDialer(addr string, auth *proxy.Auth, timeout time.Duration) *HttpConnectProxyDialer {
+	return &HttpConnectProxyDialer{
+		address: addr,
+		auth:    auth,
+		timeout: timeout,
+	}
+}
+
+type HttpConnectProxyDialer struct {
+	address string
+	auth    *proxy.Auth
+	timeout time.Duration
+}
+
+func (self *HttpConnectProxyDialer) Dial(network, addr string) (net.Conn, error) {
+	c, err := net.Dial(network, self.address)
```

Comment:
This assumes the proxy server is not TLS itself?

Reply:
One way to handle TLS vs non-TLS HTTP proxies is to require the protocol prefix on the address and inspect it:

`https://proxyhost` vs `htt://proxyhost` and then do the right thing.

Reply:
Yeah, this is just to get the bare minimum working. If/when we get more feature requests we can look at how to expand this out